dotCMS/src/main/java/com/dotcms/graphql/business/ContentAPIGraphQLFieldsProvider.java

```diff
@@ -0,0 +1,130 @@
+package com.dotcms.graphql.business;
+
+import static com.dotcms.graphql.util.TypeUtil.BASE_TYPE_SUFFIX;
+import static graphql.Scalars.GraphQLInt;
+import static graphql.Scalars.GraphQLString;
+import static graphql.schema.GraphQLFieldDefinition.newFieldDefinition;
+import static graphql.schema.GraphQLList.list;
+
+import com.dotcms.contenttype.model.type.ContentType;
+import com.dotcms.graphql.InterfaceType;
+import com.dotcms.graphql.datafetcher.ContentletDataFetcher;
+import com.dotcms.graphql.util.TypeUtil;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.util.Logger;
+import graphql.schema.GraphQLArgument;
+import graphql.schema.GraphQLFieldDefinition;
+import graphql.schema.GraphQLInterfaceType;
+import graphql.schema.GraphQLType;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.collections.PredicateUtils;
+
+/**
+ * This singleton class provides all the {@link GraphQLFieldDefinition}s needed for the Content Delivery API
+ */
+enum ContentAPIGraphQLFieldsProvider implements GraphQLFieldsProvider {
+
+    INSTANCE;
+
+    @Override
+    public Collection<GraphQLFieldDefinition> getFields() throws DotDataException {
+        // Each ContentType as query'able collection field
+        List<ContentType> contentTypeList = APILocator.getContentTypeAPI(APILocator.systemUser())
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'contentTypeList' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5642169)